main.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"fmt"
+	"go_service/src"
+	"go_service/src/adapters"
+	"go_service/src/api"
+	"go_service/src/common"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"log"
+	"net/http"
+	"os"
+)
+
+func setupHandlers(cr common.CurrencyReader, sa api.SubscriberGateway) *http.ServeMux {
+	sm := http.NewServeMux()
+	sm.HandleFunc("/", api.GetCurrencyHandler(cr))
+	sm.HandleFunc("/subscribe", api.GetSubscribersHandler(sa))
+	return sm
+}
+
+func fetchEnv(name string) string {
+	value := os.Getenv(name)
+	if value == "" {
+		log.Fatal(fmt.Sprintf("Environment variable %s is not set", name))
+	}
+	return value
+}
+
+func main() {
+	currencyRateUrl := fetchEnv("CURRENCY_RATE_URL")
+	currencyCode := fetchEnv("CURRENCY_CODE")
+	dbUser := fetchEnv("POSTGRES_USER")
+	dbPassword := fetchEnv("POSTGRES_PASSWORD")
+	dbName := fetchEnv("POSTGRES_DB")
+	dbPort := fetchEnv("DB_PORT")
+	dbHost := fetchEnv("DB_HOST")
+	email := fetchEnv("EMAIL")
+	emailPassword := fetchEnv("EMAIL_PASSWORD")
+
+	emailAdapter := adapters.EmailAdapter{Username: email, Password: emailPassword}
+	emailAdapter.CreateAuth()
```

Comment:
Why don't you move this emailAdapter creation code to a separate function in the “adapters” package that would return the object you created elsewhere instead of creating it manually in the main function?

Reply:
Adapter creation is a simple operation, i don't see reasons to place it to other factory function in this moment